subprojects/plugin-use/src/main/java/org/gradle/plugin/use/internal/DefaultPluginRequestApplicator.java

```diff
@@ -86,6 +86,9 @@ public void applyPlugins(final PluginRequests requests, final ScriptHandlerInter
 
         final PluginResolver effectivePluginResolver = wrapInAlreadyInClasspathResolver(classLoaderScope);
 
+        if (!requests.isEmpty()) {
+            addPluginArtifactRepositories(scriptHandler.getRepositories());
```

Comment:
I'm confused why this isn't a problem already...

Reply:
I think the main problem is that it interacts with the plugin repository which is only present when you declare these repo filters within the buildscripts block